### DISCUSSION TITLE
X-Forwarded-For:  headers.

### DISCUSSION THREAD

**Discussion by out-n-about** [Security-Onion-Solutions/securityonion#2751] [Unsupported Versions]
I have traffic that is going through some load balancers, break-inspect devices and various other perimeter devices.  

The source IP address gets changed on some of sessions from the actual source to the network device and then the header "X-Forwarded-For:" is added which shows that actual source address.  Other sessions it does not happen and shows the correct source IP.

Is there a way to fix the parsing so that if this header exists it replaces what it says the source address is with the address in this header field when it exists?

Thanks

**Comment by weslambert** [comment]
Do you have an example record of where this is happening so that we can work with that to consider this?

**Reply by out-n-about** [reply]
Here are entries that are from our instance. (change the identifying info)

source.ip
172.16.51.22

destination.ip
172.16.51.167

network.data.decoded
GET /cgi-bin/example.sh HTTP/1.1
Host: host.com
User-Agent: "() { :; }; echo ; /bin/bash -c 'cat /etc/passwd'"
Accept: */*
Accept-Language: en
Connection: close
X-Forwarded-For: 46.111.16.2

The "X-Forwarded-For:" is a field added by the SSL Break/Inspect appliance that shows the true source IP.  The source.ip field is changed by the tool to the IP of the break/inspect device.  This only happens when the traffic is SSL and decrypted. 

For HTTP traffic the source.ip shows the true source.

Since the source ip is not the real one for the encrypted traffic, all statistics in kibana are wrong as its not showing the correct source and indicates that the break/inspect device is all bad.

Thanks

**Reply by weslambert** [reply]
Thanks, I know Zeek has the ability to provide x-forwarded-for, but I can't remember if we enable it by default, and parsing might need to be update to display it properly in SOC or Kibana.  I'll have to look into it and get back with you.

**Reply by out-n-about** [reply]
thanks

**Comment by Stephen-Ridgway** [comment]
Hi!

Did you get an answer to this? We have the same challenge.

**Reply by Mastadamus** [reply]
Likewise. Zeek does parse out the X-Forwarded-For header properly but the IP in the Client.IP field is my load balancer not the true source IP.
